### PR TITLE
Custom + Imbalanced DataLoader 'n' torchmetrics for evaluation

### PR DESCRIPTION
**Pull request by 5uperpalo**
* I added ability to provide custom DataLoader for training or to choose between default(used previously) or imbalanced Dataloader
  * imbalanced DataLoader uses in each epoch all the samples of minority class and same number of samples from other classes with shuffling, it includes parameter to randomly oversample the minority class for possible improvement of the results, see for example https://machinelearningmastery.com/smote-oversampling-for-imbalanced-classification/
* I was missing ability to see per class metrics in binary classification task so changed the metrics to use torchmetrics
  * torchmetrics library includes all currently used metrics for evaluation and it follows almost the same logic as was used in the repository until now, ie. update the state of the metric object, compute the values, reset the state

**NOTE**
* please check the pull request if you agree with the changes I can adjust the README.MD and other notebooks to use torch metrics in their code

### CODE REVIEW

**Comment by jrzaurin** [comment]
@5uperpalo please, could you create a branch from the master branch in this repo and open a PR from there? This way I can also work on the PR as there are a number of things I want us to change. 

For the time being going to close this PR without merging
